packages/nimble-components/src/table/tests/table-keyboard-navigation.spec.ts

```diff
@@ -0,0 +1,1109 @@
+                await waitForUpdatesAsync();
+                element.focus();
+                await waitForUpdatesAsync();
+                await sendKeyPressToTable(keyArrowDown);
+            });
+
+            it('pressing Space will select the row', async () => {
+                const row = pageObject.getRow(0);
+                await sendKeyPressToTable(keySpace);
+
+                expect(row.selected).toBe(true);
+            });
+
+            it('pressing RightArrow will focus the row selection checkbox', async () => {
+                await sendKeyPressToTable(keyArrowRight);
+
+                expect(currentFocusedElement()).toBe(
+                    pageObject.getRow(0).selectionCheckbox!
+                );
+            });
+
+            it('pressing Home when a cell is focused will focus the row selection checkbox', async () => {
+                await sendKeyPressesToTable(keyArrowRight, keyArrowRight);
+
+                await sendKeyPressToTable(keyHome);
+
+                expect(currentFocusedElement()).toBe(
+                    pageObject.getRow(0).selectionCheckbox!
+                );
+            });
+
+            it('pressing End when the row selection checkbox is focused will focus the last cell', async () => {
+                await sendKeyPressesToTable(keyArrowRight);
+
+                await sendKeyPressToTable(keyEnd);
+
+                expect(currentFocusedElement()).toBe(pageObject.getCell(0, 2));
+            });
+        });
+
+        describe('for a table with a grouped column, when a group row is focused,', () => {
+            beforeEach(async () => {
+                await setupTableWithGrouping();
+                element.focus();
+                await waitForUpdatesAsync();
+                await sendKeyPressToTable(keyArrowDown);
+                if (!(currentFocusedElement() instanceof TableGroupRow)) {
+                    throw new Error('Expected group row to be focused');
+                }
+            });
+
+            describe('for an expanded group row, pressing the given key will collapse the group:', () => {
+                const tests = [
+                    { name: 'Space', key: keySpace },
+                    { name: 'Enter', key: keyEnter },
+                    { name: 'LeftArrow', key: keyArrowLeft }
+                ];
+                parameterizeSpec(tests, (spec, name, value) => {
+                    spec(name, async () => {
+                        await sendKeyPressToTable(value.key);
+
+                        const focusedElement = currentFocusedElement();
+                        expect(focusedElement).toBeInstanceOf(TableGroupRow);
+                        expect((focusedElement as TableGroupRow).expanded).toBe(
+                            false
+                        );
+                    });
+                });
+            });
+
+            describe('for a collapsed group row, pressing the given key will expand the group:', () => {
```

Comment:
I think we're missing equivalent tests for collapsing/expanding hierarchical rows (or I missed them)

Reply:
Added those tests.

Reply:
Added